pkg/cloud/services/s3/s3.go

```diff
@@ -78,6 +79,10 @@ func (s *Service) ReconcileBucket() error {
 		return errors.Wrap(err, "ensuring bucket policy")
 	}
 
+	if err := s.ensureBucketLifecycleConfiguration(bucketName); err != nil {
```

Comment:
Main change number 2 (including the IAM permission; note how it was already present in [our CAPA IAM role](https://github.com/giantswarm/giantswarm-aws-account-prerequisites/blob/master/capa-controller-role/capa-controller-policy.json))